app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -166,5 +170,60 @@ def kubelet_contents
       }
     }
   end
+
+  def generate_user_block(con)
+    user = {}
+    user[:base_dn] = con.user_base_dn
+    user[:filter]  = con.user_filter
+    user_attr_map = user[:attr_map] = {}
+    user_attr_map[:username] = con.user_attr_username
+    user_attr_map[:id]       = con.user_attr_id
+    user_attr_map[:email]    = con.user_attr_email
+    user_attr_map[:name]     = con.user_attr_name
+    user
+  end
+
+  def generate_bind_block(con)
+    bind = {}
+    if con.bind_anon
+      bind[:anonymous] = true
+    else
+      bind[:anonymous] = false
+      bind[:dn] = con.bind_dn
+      bind[:pw] = con.bind_pw
+    end
+    bind
+  end
+
+  def generate_group_block(con)
```

Comment:
can we put `generate_group_block`, `generate_bind_block`, `generate_user_block` into a library e.g. under either `lib/velum/dex/ldap.rb`, `lib/velum/dex_connectors.rb` or even to https://github.com/kubic-project/velum/blob/master/lib/velum/ldap.rb ?

i think we can expect more connectors to be added soon (SAML e.g.), therefore we could put the dex connector business logic already in a good place for not having to move it again later, what do you think?

Reply:
Agreed that it would be cleaner to have the various dex related code pulled out and be separate. It does make sense as well to place them in different files per type of connector, for ease of navigating to the relevant code. I will split it up that way and update this PR.

Yes, we are working on adding OIDC and SAML support as well.